From f80148e2aa630fdc957628ef47fe121393da57be Mon Sep 17 00:00:00 2001
From: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Date: Wed, 13 Mar 2013 12:08:11 -0700
Subject: Dalvik: Accept More Loops for Registerization Part 1

BZ: 92848

- Removes the restriction of number of BasicBlocks on the loops being
    registerized

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ia9f37ad0282ad2b388ef493915b48942688e971d
Orig-MCG-Change-Id: I32358c6a15f3d2ebae63012b23862a5ef4e88137
Signed-off-by: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/RegisterizationME.cpp |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/vm/compiler/RegisterizationME.cpp b/vm/compiler/RegisterizationME.cpp
index 41b5373..4e3c9e2 100644
--- a/vm/compiler/RegisterizationME.cpp
+++ b/vm/compiler/RegisterizationME.cpp
@@ -564,12 +564,6 @@ static bool checkLoop (const CompilationUnit *cUnit, const LoopInformation *info
 
     int elems = dvmCountSetBits (blocks);
 
-    //If more than one block, quit
-    if (elems > 2)
-    {
-        return false;
-    }
-
     //Go through each block
     BitVectorIterator bvIterator;
 
-- 
1.7.4.1

